sdcm/utils/uda.py

```diff
@@ -0,0 +1,109 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2022 ScyllaDB
+from pathlib import Path
+from typing import Optional
+
+import yaml
+from pydantic import BaseModel
+
+from sdcm.utils.udf import UDF, UDFS
+
+
+class UDA(BaseModel):
+    """
+    Provides the representation for User Defined Aggregates in SCT,
+    and the interface for creating queries for them.
+
+    User Defined Aggregates work very much the same as regular CQL aggregate functions,
+    and can be defined via CQL, e.g. :
+
+        CREATE FUNCTION accumulate_len(acc tuple<bigint,bigint>, a text)
+        RETURNS NULL ON NULL INPUT
+        RETURNS tuple<bigint,bigint>
+        LANGUAGE lua as 'return {acc[1] + 1, acc[2] + #a}';
+
+        CREATE OR REPLACE FUNCTION present(res tuple<bigint,bigint>)
+        RETURNS NULL ON NULL INPUT
+        RETURNS text
+        LANGUAGE lua as 'return "The average string length is " .. res[2]/res[1] .. "!"';
```

Comment:
Please add info that corresponding UDF's used in UDA must be created before creating UDA.
Maybe add this info to `get_create_query_string` method?

Reply:
I've added it to the class docstring. That should be enough. Trying to create the UDA without creating the UDFs first will result in a sensible error message from Scylla about being unable to find the referenced UDF function.